### ISSUE TITLE
DataTable/Table not able to navigate to data cells off-screen with TalkBack accessibility focus

### ISSUE THREAD

**Issue by harperl-lgtm** [flutter/flutter#112957]
## Steps to Reproduce

1. Execute `flutter run` to run flutter gallery on your phone / emulator (android) https://github.com/flutter/gallery 
2. Click 'MATERIAL' 
3. Click 'DataTable' 
4. Turn on TalkBack and navigate through the data table

**Expected results:** 
Should be able to autoscroll and move accessibility focus accordingly as we navigate to data cells off-screen

**Actual results:** 
No autoscroll and TalkBack focus ignores all data cells off-screen.

<details>
<summary>Code sample</summary>

```
Related Code files:
https://github.com/flutter/gallery/blob/main/lib/demos/material/data_table_demo.dart
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/data_table.dart
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/table.dart
```

</details>

<details>
  <summary>Logs</summary>

```
```

```
```

```
```

</details>

**Comment by chunhtai** [comment]
It looks like the vertical scrolling works but not the horizontal scrolling. The horizontal scrollable child is not a sliver, so it need to rely on showonscreen to auto scroll, there may be some problem with the showonscreen code.

**Comment by harperl-lgtm** [comment]
Update:

Looks like the Gallery demo uses a PaginatedDataTable, which uses DataTable under the hood and uses a SingleChildScrollView to control the horizontal scrollability of the table.

Calling Scrollable.ensureVisible() in OnDidGainAccessibilityFocus method of the DataCells was able to make the screen autoscroll horizontally when navigating forward (left to right) and backward (right to left) within a row. This works likely because in our case the DataCells are relatively small and within the same row the next cell is always at least partially visible and thus was able to gain accessibility focus, and then Scrollable.ensureVisible() will then scroll it to the center.

In the case where we are at the last (rightmost) DataCell in a row and the logical next is the first (leftmost) DataCell in the next row, the first cell in the next row is always invisible so even with Scrollable.ensureVisible() the screen does not autoscroll to the left because the next cell is completely invisible not able to gain accessibility focus in the first place.

**Comment by HansMuller** [comment]
CC @piinks

**Comment by Piinks** [comment]
The DataTable and Table widgets in the framework are not scrollable, and are subject to these kind of issues when trying to place them inside of multiple scrollables for a 2D effect. We have the TableView now (https://pub.dev/packages/two_dimensional_scrollables) which resolves this issue!
